eth/filters/filter_system.go

```diff
@@ -626,7 +641,9 @@ func (es *EventSystem) lightFilterLogs(header *types.Header, addresses []common.
 		// Get the logs of the block
 		ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
 		defer cancel()
-		logsList, err := es.sys.cachedGetLogs(ctx, header.Hash(), header.Number.Uint64())
+		indexed := es.sys.backend.LastBloomIndex()
+		headerNumber := header.Number.Uint64()
+		logsList, err := es.sys.cachedGetLogs(ctx, header.Hash(), headerNumber, indexed > headerNumber)
```

Comment:
Noticing we could actually remove `lightFilterLogs` completely, out of scope for this PR though

Reply:
Also no real need to

Reply:
Yeah, I thought the same.